docs/pywbemcli/commands.rst

```diff
@@ -1534,6 +1549,172 @@ See :ref:`pywbemcli server remove-mof --help` for the exact help output of the
 command.
 
 
+.. index:: pair: command groups;statistics command group
+
+.. _`Statistics command group`:
+
+``statistics`` command group
+----------------------------
+
+The ``statistics`` command group includes commands that manage and display
+statistics on the WBEM server operations executed by pywbemcli.
+
+Statistics can be gathered in the pywbemcli client itself and also separately
+by some WBEM servers.
+
+These servers provide statistical information on the overall server
+time that can be reported to the client with each operation response.  The
+enabling/disabling of client statistics and server statistics are separate.
+
+Client statistics gathering is enabled with the pywbemcli general option
+``--timestats`` / ``-T`` including  number of operations
+executed for each operation type, operation execution time (mininum, maximum, and
+average), minimum, maximum, and average size of requests and responses, and
+the server response time if it is enabled on the server.
```

Comment:
No. It also controls pywbemcli gathering.  How about 

u'Enable time statistics gathering in pywbemcli of '
              u'CIM/XML operations and define when displayed: '

Reply:
If we really remove the choice 'none' this means the default cannot stay 'none' or some disabled state. The default must be one of the valid choices, and both remaining choices have the gathering enabled. The client-side gathering is now just always enabled, and IMO it is not a problem to do that. We talk about 20+ integer values, that should not be an issue in a Python based command that consumes some MB.

Reply:
Why can't we have the default None.  That is what is applied if the option is not defined.  Note that we do not make cmd or auto the default, just the fact that we do not gather any statistics.

Since we do not have a default none, there is no way to disable statistics in the interactive mode once it is enabled. But I do not see that as an issue since setting it to the cmd mode eliminates the display and as you said, this is probably not a major resource utilizer.

Reply:
**DISCUSSION**

Reply:
For the timesats the help states now that  "Default: . EnvVar PYWBEMCLI_TIMESTATS, or None

The objection is that None is not one of the choices so does not represent a real default.

We use default in other general opotions as folows

1. --use-pull - choice option where the default is one of the choices
2. --pull-count - Integer value where the default is a specific integer count
3. --user - string option where default is EnvVar or none
4. --password where defaults is envvar or prompted if --user exists
.

Generally, Default in the help means:

1. For a string or integer  option; A value that will be applied if the option is not included in the command.

2. For a choice item; that value of the choice that will be applied if the option is not included in the command

3. For a boolean - Since we are . doing the booleans as --xxx / --no-xxx it should represent the one of the two options that apply if the option is not included.  NOTE: We do not really do that.  Look at the following:

  --warn / --no-warn              Warnings control: True enables display of all
                                  Python warnings; False leaves warning control
                                  to the PYHONWARNINGS env var, which by default
                                  displays no warnings. Default: False.
 
It is not clear how --warn and --no-warn relate to True and False

In this case, we have two choices:I think.

Today --timestats does two things:

1. Enable statistics if the option is set
2. Determine when the statistics are displayed.

I think the choices are:

1. Add none back in as a choice and make that the default. That is what Andy originally objected to.

2. Change the characteristics of the pybemcli code to always enable statistics gathering and make the --timestats option simply set what is now the auto mode. The timestats could be displayed/reset at any time  (statistics show) and pywbemcli would always be actively gathering the data defined in the pywbem statistics class. Since this would return timestats to a boolean, we would then change it to --timestats, --no-timestats so the user could enable/disable the auto display at in interactive mode.

The only purpose of the --timestats option would be to set what is now the auto mode, in effect where we started. 

**DISCUSSION**

Reply:
Modified code to make --timestats boolean and always enable statistics.  Statistics always displayed when command "statistics show" entered and displayed after each command when general options --timestats entered.  This also adds the --no-timestats string to the option so the display can be turned on and off in interactive mode

Reply:
I think this is a good decision.